consensus/src/epoch_manager.rs

```diff
@@ -334,18 +337,143 @@ impl EpochManager {
+            let block_store = Arc::new(BlockStore::new(
+                Arc::clone(&self.storage),
+                recovery_data,
+                self.state_computer.as_ref().unwrap().clone(),
+                self.config.max_pruned_blocks_in_mem,
+                Arc::clone(&self.time_service),
+            ));
+
+            info!(epoch = epoch, "Create ProposalGenerator");
+            // txn manager is required both by proposal generator (to pull the proposers)
+            // and by event processor (to update their status).
+            let proposal_generator = ProposalGenerator::new(
+                self.author,
+                block_store.clone(),
+                self.txn_manager.clone(),
+                self.time_service.clone(),
+                self.config.max_block_size,
+            );
+
+            let guage_c = IntGauge::new(
+                "D_COM_CHANNEL_COUNTER_EM",
+                "counter for the decoupling committing channel in epoch manager",
+            )
+            .unwrap();
+
+            let (comm_tx, comm_rx) = channel::new::<(
+                Vec<ExecutedBlock>,
+                LedgerInfoWithSignatures,
+            )>(self.config.channel_size, &guage_c);
+
+            let execution_phase = ExecutionPhase::new(
+                exec_rx,
+                self.commit_channel_state_computer.clone(),
+                comm_tx,
+            );
+
+            tokio::spawn(execution_phase.start());
+
+            let guage_c_msg = IntGauge::new(
+                "D_COM_CHANNEL_COUNTER_EM",
+                "counter for the decoupling committing channel in epoch manager",
+            )
+            .unwrap();
+            let (commit_msg_tx, commit_msg_rx) =
+                channel::new::<ConsensusMsg>(self.config.channel_size, &guage_c_msg);
+
+            self.sender_commit_msg = Some(commit_msg_tx);
+
+            self.back_pressure
+                .store(0, std::sync::atomic::Ordering::SeqCst);
+
+            let commit_phase = CommitPhase::new(
+                comm_rx,
+                self.commit_channel_state_computer.clone(),
+                commit_msg_rx,
+                epoch_state.verifier.clone(),
+                Arc::clone(&safety_rules_container),
+                self.author,
+                self.back_pressure.clone(),
+                network_sender.clone(),
+            );
+
+            tokio::spawn(commit_phase.start());
+
+            RoundManager::new(
+                epoch_state,
+                block_store,
+                round_state,
+                proposer_election,
+                proposal_generator,
+                Arc::clone(&safety_rules_container),
+                network_sender,
+                self.txn_manager.clone(),
+                self.storage.clone(),
+                self.config.sync_only,
+                self.back_pressure.clone(),
+                true,
+                self.config.back_pressure_limit,
+            )
```

Comment:
this is too long, we should split into smaller functions and reuse some of the code snippet and have round manager to have new and new_ordering_only to accept some optional arguments like backpressure

Reply:
resolved